### PR TITLE
Support revive for go files

### PR DESCRIPTION
**Pull request by liaoishere**
[revive](https://github.com/mgechev/revive) is a drop-in replacement of golint.
It is extensible, configurable, provides a superset of the rules of golint, and has significantly better performance.

### CODE REVIEW

**Comment by aymericbeaumet** [comment]
I'm really interested in seeing this merged upstream.

I've been using it for some time now (by appending [this snippet](https://github.com/aymericbeaumet/dotfiles/blob/031d18531615c6027e3327f6030d370da0989d89/src/.vimrc#L271-L286) to my .vimrc) and it's working like a charm.

**Comment by liaoishere** [comment]
ping @w0rp

File: ale_linters/go/revive.vim [hunk 1]
```diff
@@ -0,0 +1,21 @@
+" Author: Penghui Liao <liaoishere@gmail.com>
+" Description: Adds support for revive
+
+call ale#Set('go_revive_executable', 'revive')
+call ale#Set('go_revive_options', '')
+
+function! ale_linters#go#golint#GetCommand(buffer) abort
```
Comment (RyanSquared) [inline ale_linters/go/revive.vim:7]: please rewrite this as ale_linters#go#revive#GetCommand

File: ale_linters/go/revive.vim [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    return ale#go#EnvString(a:buffer) . '%e'
+    \   . (!empty(l:options) ? ' ' . l:options : '')
+    \   . ' %t'
+endfunction
+
+call ale#linter#Define('go', {
+\   'name': 'revive',
+\   'output_stream': 'both',
+\   'executable': {b -> ale#Var(b, 'go_revive_executable')},
+\   'command': function('ale_linters#go#golint#GetCommand'),
```
Comment (RyanSquared) [inline ale_linters/go/revive.vim:19]: and replace this as well, this would still call the original golang linter.

**Comment by RyanSquared** [comment]
@liaoishere your current work looks like it would cause conflict with the original golang linter. please address my comments and i can probably get this merged in.

**Comment by liaoishere** [comment]
> @liaoishere your current work looks like it would cause conflict with the original golang linter. please address my comments and i can probably get this merged in.

Thx for review. All the comments are resolved.

**Comment by liaoishere** [comment]
ping @RyanSquared

File: doc/ale-go.txt [hunk 3]
```diff
@@ -219,6 +219,25 @@ g:ale_go_govet_options                                 *g:ale_go_govet_options*
   This variable can be set to pass additional options to the go vet linter.
 
 
+===============================================================================
+revive                                                          *ale-go-revive*
+
+g:ale_go_reivve_executable                         *g:ale_go_revive_executable*
```
Comment (aymericbeaumet) [inline doc/ale-go.txt:225]: `g:ale_go_reivve_executable` -> `g:ale_go_revive_executable`

Comment (aymericbeaumet) [reply doc/ale-go.txt:225]: @liaoishere Just found a small typo when re-reading the PR

Comment (liaoishere) [reply doc/ale-go.txt:225]: nice catch, I’ll fix it

**Comment by RyanSquared** [comment]
Cheers! :beers:

**Pull request merged**